crates/ere-risc0/src/prove/bento.rs

```diff
@@ -0,0 +1,230 @@
+            }
+            "SUCCEEDED" => {
+                let receipt_bytes = client
+                    .receipt_download(&session)
+                    .map_err(|err| zkVMError::Other(err.into()))?;
+                break Ok((bincode::deserialize(&receipt_bytes).unwrap(), now.elapsed()));
+            }
+            _ => {
+                return Err(zkVMError::Other(
+                    format!("Unexpected proving status: {}", res.status).into(),
+                ));
+            }
+        }
+    }
+}
+
+fn cmd_output_checked(cmd: &mut Command) -> Result<Output, io::Error> {
+    let output = cmd.output()?;
+    if !output.status.success() {
+        return Err(io::Error::other(format!("Failed to run `{cmd:?}`")));
+    }
+    Ok(output)
+}
+
+fn cmd_exec_checked(cmd: &mut Command) -> Result<(), io::Error> {
+    let status = cmd.status()?;
+    if !status.success() {
+        return Err(io::Error::other(format!("Failed to run `{cmd:?}`")));
+    }
+    Ok(())
+}
+
+fn docker_image_exists(image: impl AsRef<OsStr>) -> Result<bool, io::Error> {
+    let output = cmd_output_checked(
+        Command::new("docker")
+            .args(["images", "--quiet"])
+            .arg(image),
+    )?;
+    // If image exists, image id will be printed hence stdout will be non-empty.
+    Ok(!output.stdout.is_empty())
+}
+
+fn docker_image_tag(src: impl AsRef<OsStr>, dst: impl AsRef<OsStr>) -> Result<(), io::Error> {
+    cmd_exec_checked(
+        Command::new("docker")
+            .args(["image", "tag"])
+            .arg(src)
+            .arg(dst),
+    )
+}
+
+pub fn build_bento_images() -> Result<(), io::Error> {
+    let agent_tag = format!("ere-risc0/agent:{SDK_VERSION}");
+    let rest_api_tag = format!("ere-risc0/rest_api:{SDK_VERSION}");
+
+    if docker_image_exists(&agent_tag)? && docker_image_exists(&rest_api_tag)? {
+        return Ok(());
+    }
+
+    let tempdir = tempdir()?;
+
+    cmd_exec_checked(
+        Command::new("git")
+            .args([
+                "clone",
+                "--single-branch",
+                "--branch",
+                &format!("v{SDK_VERSION}"),
+                "https://github.com/risc0/risc0.git",
+            ])
+            .arg(tempdir.path()),
+    )?;
+
+    cmd_exec_checked(
+        Command::new("docker")
+            .arg("compose")
+            .arg("--file")
+            .arg(tempdir.path().join("compose.yml"))
+            .arg("--env-file")
+            .arg(tempdir.path().join("bento/dockerfiles/sample.env"))
+            .arg("build"),
+    )?;
+
+    docker_image_tag("agent", agent_tag)?;
+    docker_image_tag("bento-rest_api", rest_api_tag)?;
+
+    Ok(())
+}
+
+const BENTO_COMPOSE_TEMPLATE: &str = include_str!("../../compose.yml");
+const BENTO_GPU_PROVER_AGENT_TEMPLATE: &str = include_str!("../../gpu_prover_agent.yml");
+
+fn bento_compose_file() -> String {
+    let cuda_visible_devices = env::var("CUDA_VISIBLE_DEVICES").unwrap_or_else(|_| "".to_string());
```

Comment:
If `ere-risc0` is run natively, this means the client must set this env var accordingly?

Reply:
Yes, it'd be ideal to set this for ere to know how many prover agent container to spin up.

If ran by `ere-dockerized` we also need to set this (and it'd be inherited properly)